lib/src/elastic_client_impl.dart

```diff
@@ -190,7 +190,12 @@ class Client {
     Map aggregations,
     Duration scroll,
   }) async {
-    final path = [index, type, '_search'];
+    var path = [index, '_search'];
```

Comment:
Let's use the following format:

```
final path = [
  index,
  if (type != null) type,
  '_search',
];
```

Reply:
The intention here was really to provide a short term backward compatible change. I was thinking about annotating the parameters with `@Deprecated` so it can be removed on the next major. Unfortunately, with this approach, you are still required to pass the parameter as null, and that would have to be manually removed in the next major. The only way the maybe the user could avoid the bc break would be providing the type as a named argument. Dart requirement to put the optional parameters at the end, and lack of method overloading, makes it really hard to prepare for future versions.

> Let's use the following format:
> 
> ```
> final path = [
>   index,
>   if (type != null) type,
>   '_search',
> ];
> ```

Isn't that going to declare the second parameter anyway?

The intention here is really to produce a list of two items, the first being the index and the second being the '_search', so the end result is 'hostname:9200/index/_search'. (as a side note, in my elastic search setup I wasn't even allowing the type param to be provided and had to change just because of the lack of support here).

I also intentionally used the code style like that. In my opinion, it is **more readable** to not have the assignment mixed with the condition.

Another option would have been 

```dart
final path = type ? [index, type, '_search'] : [index, '_search'];
```

The final/var switch didn't seem important enough to make it slightly harder to read.

Anyway, you are the owner, so it's up to you to decide the code style. After consideration, please confirm or go ahead and change it directly.

Reply:
> Isn't that going to declare the second parameter anyway (as null?)?

No, that's the point of the inlined conditional element: if it is null, it will be left out.

> final path = type ? [index, type, '_search'] : [index, '_search'];

In effect it is equivalent of the single list creation with the conditional.